flipt-engine-ffi/src/lib.rs

```diff
@@ -97,31 +99,32 @@ impl Default for EngineOpts {
 }
 
 pub struct Engine {
-    pub opts: EngineOpts,
-    pub evaluator: Arc<Mutex<Evaluator<HTTPParser, Snapshot>>>,
+    fetcher: HTTPFetcher,
+    evaluator: Arc<Mutex<Evaluator<Snapshot>>>,
+    runtime: Runtime,
+    _update_handle: JoinHandle<()>,
 }
 
 impl Engine {
-    pub fn new(evaluator: Evaluator<HTTPParser, Snapshot>, opts: EngineOpts) -> Self {
-        let mut engine = Self {
-            opts,
-            evaluator: Arc::new(Mutex::new(evaluator)),
-        };
-
-        engine.update();
-
-        engine
-    }
+    pub fn new(mut fetcher: HTTPFetcher, evaluator: Evaluator<Snapshot>) -> Self {
+        let runtime = Runtime::new().expect("Failed to create Tokio runtime");
+        let evaluator = Arc::new(Mutex::new(evaluator));
+        let evaluator_clone = evaluator.clone();
 
-    fn update(&mut self) {
-        let evaluator = self.evaluator.clone();
-        let update_interval = self.opts.update_interval.unwrap_or(120);
+        let mut rx = runtime.block_on(async { fetcher.start() });
 
-        std::thread::spawn(move || loop {
-            std::thread::sleep(std::time::Duration::from_secs(update_interval));
-            let mut lock = evaluator.lock().unwrap();
-            lock.replace_snapshot();
+        let update_handle = runtime.spawn(async move {
+            while let Some(res) = rx.recv().await {
+                evaluator_clone.lock().unwrap().replace_snapshot(res);
+            }
```

Comment:
I think it is worth doing an initial replace_snapshot before spawning this thread to ensure at-least one snapshot loads. This may be the source of empty evals seen in ITs.

Reply:
that was it!